### PR TITLE
Use the govuk_sidekiq gem

### PR DESCRIPTION
**Pull request by mgrassotti**
https://trello.com/c/JQREOuvh/424-use-sidekiq-gem

Use the govuk gem to setup and initialize sidekiq.
The headers inclusion in the worker args is no longer required as it is
already performed inside the gem.
The redis version had to be increased to comply with the new gem
requirements.

### CODE REVIEW

**Comment by mgrassotti** [comment]
The test is failing, probably because the header `{'Govuk-Request-Id'=>'12345-67890'}` is not present in the request `PUT http://publishing-api.dev.gov.uk/v2/content/[...]`.
I guess the header should be included by the gem with the worker `perform` action, but it seems not.
The trace follows:
```
RequestTracingTest
  test_govuk_request_id_is_passed_downstream_across_the_worker_boundary_on_publish FAIL (0.29s)
Minitest::Assertion:         The request PUT /publishing-api.*content\/d86e3b3c-ff6c-4d74-b928-b7188eecea7f/ with headers {'Govuk-Request-Id'=>'12345-67890'} was expected to execute 7 times but it executed 0 times
        
        The following requests were made:
        
        PUT http://panopticon.dev.gov.uk/artefacts/.json with body '{"slug":null,"owning_app":"publisher","kind":"answer","name":"a title","description":null,"state":"live","rendering_app":"frontend","sections":[],"paths":["/.json"],"prefixes":["/"],"specialist_sectors":[null],"indexable_content":"","public_timestamp":"2016-06-22T12:18:03.318+01:00","latest_change_note":null}' with headers {'Accept'=>'application/json', 'Accept-Encoding'=>'gzip, deflate', 'Authorization'=>'Bearer set_at_deploy_time', 'Content-Length'=>'310', 'Content-Type'=>'application/json', 'Govuk-Request-Id'=>'12345-67890', 'User-Agent'=>'gds-api-adapters/30.2.1 ()'} was made 1 time
        PUT http://publishing-api.dev.gov.uk/v2/content/d86e3b3c-ff6c-4d74-b928-b7188eecea7f with body '{"title":"a title","base_path":"/","description":"","schema_name":"placeholder","document_type":"answer","need_ids":[],"public_updated_at":"2016-06-22T12:18:03.318+01:00","publishing_app":"publisher","rendering_app":"frontend","routes":[{"path":"/","type":"exact"}],"redirects":[],"update_type":"major","details":{"change_note":null,"tags":{"browse_pages":[],"primary_topic":[],"additional_topics":[],"topics":[]}},"locale":"en"}' with headers {'Accept'=>'application/json', 'Accept-Encoding'=>'gzip, deflate', 'Authorization'=>'Bearer example', 'Content-Length'=>'429', 'Content-Type'=>'application/json', 'User-Agent'=>'gds-api-adapters/30.2.1 ()'} was made 7 times
        POST http://publishing-api.dev.gov.uk/v2/content/d86e3b3c-ff6c-4d74-b928-b7188eecea7f/publish with body '{"update_type":{"authenticated_user":null,"request_id":"12345-67890"}}' with headers {'Accept'=>'application/json', 'Accept-Encoding'=>'gzip, deflate', 'Authorization'=>'Bearer example', 'Content-Length'=>'70', 'Content-Type'=>'application/json', 'User-Agent'=>'gds-api-adapters/30.2.1 ()'} was made 1 time
        PATCH http://publishing-api.dev.gov.uk/v2/links/d86e3b3c-ff6c-4d74-b928-b7188eecea7f with body '{"links":{"mainstream_browse_pages":[],"parent":[],"topics":[]}}' with headers {'Accept'=>'application/json', 'Accept-Encoding'=>'gzip, deflate', 'Authorization'=>'Bearer example', 'Content-Length'=>'64', 'Content-Type'=>'application/json', 'User-Agent'=>'gds-api-adapters/30.2.1 ()'} was made 1 time
        
        ============================================================
        test/integration/request_tracing_test.rb:71:in `block (2 levels) in <class:RequestTracingTest>'
        test/integration/request_tracing_test.rb:13:in `block in <class:RequestTracingTest>'
```

**Comment by tijmenb** [comment]
@mgrassotti that is probably caused by Sidekiq not using middleware during testing:

https://github.com/mperham/sidekiq/wiki/Testing#testing-server-middleware

**Comment by mgrassotti** [comment]
👍 great @tijmenb thanks!

**Comment by mgrassotti** [comment]
To be updated to use the latest version of the govuk_sidekiq gem

File: Gemfile [hunk 1]
```diff
@@ -52,7 +52,7 @@ gem 'null_logger'
 gem 'plek', '1.9.0'
 gem 'diffy', '3.0.6'
 
-gem 'redis', '3.0.7', require: false # Only used in some importers
+gem 'redis', '3.2.1', require: false # Only used in some importers
```
Comment (boffbowsh) [inline Gemfile]: I don't think we need to explicitly require / pin this. It's required by Sidekiq and `mlanett-redis-lock`, both of which are explicitly required here.

Comment (mgrassotti) [inline Gemfile]: 👍 ok

**Comment by boffbowsh** [comment]
LGTM, merge on Thursdasy 👍

**Comment by mgrassotti** [comment]
👍 this also needs https://github.com/alphagov/govuk-puppet/pull/4583

**Comment by tijmenb** [comment]
https://github.com/alphagov/publisher/blob/master/config/redis.yml should also be removed.

**Comment by mgrassotti** [comment]
Thanks @tijmenb, done in https://github.com/alphagov/publisher/pull/488

**Pull request merged**